### ISSUE TITLE
Singleton vs several instances

### ISSUE THREAD

**Issue by ZeeCoder** [que-etc/resize-observer-polyfill#13]
Hi!

I've decided to use this polyfill in my library, and I was wondering which would be more sensible:
1) creating a single observer, and use that to observe all elements,
2) creating a new instance for each element I want to observe.

More context:
I'm making a container query solution, in which I instantiate a "Container" class on each element that supports container queries.
In the module where the class is declared, I could create a new ResizeObserver instance for each element, or I could create a single resize observer, and just call observe() on that for each new element.

The difference though, is if I want to use a singleton, then I'll need to map elements to Container instances, since on each change I'll want to call an `adjust` method.

In summary: my dilemma is whether it's worth implementing such an Element => Container map, that the singleton resize observer would use, or should I just create a new instance for each new Container, which would have access to the aforementioned `adjust` method by default.

(WIP: https://github.com/ZeeCoder/container-query)

**Comment by que-etc** [comment]
Hi.

The main downside of having multiple observers and the reason why you might want to stick with the singleton approach, is that there might be a conflict between RO instances, so to speak. This may happen when one of the RO callbacks affects the dimensions of elements that are tracked by another observer which in turn may lead to a discrepancy between the reported dimensions and the actual ones. A [sample](https://jsfiddle.net/que_etc/tcs6n9ku/) to illustrate it.

As for the rest of it, the difference between approaches is somewhat marginal and in most cases I've seen, if not in all of them, people are quiet nonchalant and don't bother themselves with keeping only one RO instance. Though it's certainly a good thing that you do :)

**Comment by ZeeCoder** [comment]
I had a hunch that it's the right approach. Thanks for the feedback!
